### PR TITLE
test: user data in function property descriptor

### PR DESCRIPTION
**Pull request by KevinEady**
Adds custom user-data tests to object property descriptors of function type.

### CODE REVIEW

File: test/object/object.cc [hunk 1]
```diff
@@ -77,7 +82,9 @@ void DefineProperties(const CallbackInfo& info) {
 
   Boolean trueValue = Boolean::New(env, true);
   UserDataHolder* holder = new UserDataHolder();
+  UserDataHolder* functionHolder = new UserDataHolder();
```
Comment (legendecas) [inline test/object/object.cc]: I think its safe to use the `holder` above rather than creating a new one.

Comment (KevinEady) [reply test/object/object.cc]: Yeah, I thought about that, but... didn't :smile: ... Addressed

**Comment by gabrielschulhof** [comment]
CI:

| Version | Job | Status |
|---:|---|:---:|
| v14.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1659/ | ✓ |
| v12.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1660/ | ✓ |
| v10.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1661/ | ✓ |

**Comment by gabrielschulhof** [comment]
Landed in 46484202caf180d7c16734b6c5a3f0cc88ee6dd0.

**Pull request closed without merging**